core/src/main/java/com/linecorp/armeria/client/ClientFactoryBuilder.java

```diff
@@ -419,36 +428,44 @@ public ClientFactoryBuilder idleTimeoutMillis(long idleTimeoutMillis) {
+     * a <a href="https://httpwg.org/specs/rfc7540.html#PING">PING</a> frame is sent for HTTP/2 or
+     * an <a herf="https://tools.ietf.org/html/rfc7231#section-4.3.7">OPTIONS</a> request with an asterisk ("*")
+     * is sent for HTTP/1.
      *
-     * @param http2PingTimeoutMillis the timeout in milliseconds. {@code 0} disables the timeout.
-     */
-    public ClientFactoryBuilder http2PingTimeoutMillis(long http2PingTimeoutMillis) {
-        checkArgument(http2PingTimeoutMillis >= 0,
-                      "http2PingTimeoutMillis: %s (expected: >= 0)", http2PingTimeoutMillis);
-        option(ClientFactoryOption.HTTP2_PING_TIMEOUT_MILLIS, http2PingTimeoutMillis);
-        return this;
-    }
-
-    /**
-     * Sets the HTTP/2 <a href="https://httpwg.org/specs/rfc7540.html#PING">PING</a> timeout.
+     * <p>Note that this settings is only in effect when {@link #idleTimeoutMillis(long)}} or
+     * {@link #idleTimeout(Duration)} is greater than the specified PING interval.
      *
-     * @param http2PingTimeout the timeout. {@code 0} disables the timeout.
-     */
-    public ClientFactoryBuilder http2PingTimeout(Duration http2PingTimeout) {
-        requireNonNull(http2PingTimeout, "http2PingTimeout");
-        checkArgument(http2PingTimeout.toMillis() >= 0,
-                      "http2PingTimeoutMillis: %s (expected: >= 0)", http2PingTimeout.toMillis());
-        http2PingTimeoutMillis(http2PingTimeout.toMillis());
+     * <p>The minimum PING interval is 10 seconds. {@code 0} means the client will not send a PING.
+     *
+     * @throws IllegalArgumentException if the specified {@code pingIntervalMillis}
+     *                                  is smaller than 10000 milliseconds.
+     */
+    public ClientFactoryBuilder pingIntervalMillis(long pingIntervalMillis) {
+        checkArgument(pingIntervalMillis == 0 || pingIntervalMillis >= MIN_PING_INTERVAL_MILLIS,
+                      "pingIntervalMillis: %s (expected: >= %s or == 0)", pingIntervalMillis,
+                      MIN_PING_INTERVAL_MILLIS);
+        option(ClientFactoryOption.PING_INTERVAL_MILLIS, pingIntervalMillis);
         return this;
     }
 
     /**
-     * Sets whether to send HTTP/2 <a href="https://httpwg.org/specs/rfc7540.html#PING">PING</a>
-     * when there are no active streams open.
+     * Sets the PING interval.
+     * When neither read nor write was performed for the given {@code pingInterval},
+     * a <a href="https://httpwg.org/specs/rfc7540.html#PING">PING</a> frame is sent for HTTP/2 or
+     * an <a herf="https://tools.ietf.org/html/rfc7231#section-4.3.7">OPTIONS</a> request with an asterisk ("*")
+     * is sent for HTTP/1.
+     *
+     * <p>Note that this settings is only in effect when {@link #idleTimeoutMillis(long)}} or
+     * {@link #idleTimeout(Duration)} is greater than the specified PING interval.
+     *
+     * <p>The minimum PING interval is 10 seconds. {@code 0} means the client will not send a PING.
```

Comment:
This could be
`<p>The minimum PING interval is {@value #MIN_PING_INTERVAL_MILLIS} milliseconds.`